data/src/main/java/com/ericsson/bss/cassandra/ecchronos/data/sync/EccNodesSync.java

```diff
@@ -0,0 +1,275 @@
+    private static final String TABLE_NAME = "nodes_sync";
+
+    private final CqlSession mySession;
+    private final List<Node> myNodesList;
+    private final String ecChronosID;
+
+    private final PreparedStatement myCreateStatement;
+    private final PreparedStatement myUpdateStatusStatement;
+
+    private EccNodesSync(final Builder builder) throws UnknownHostException
+    {
+        mySession = Preconditions.checkNotNull(builder.mySession, "Session cannot be null");
+        myNodesList = Preconditions
+                .checkNotNull(builder.initialNodesList, "Nodes list cannot be null");
+        myCreateStatement = mySession.prepare(QueryBuilder.insertInto(KEYSPACE_NAME, TABLE_NAME)
+                .value(COLUMN_ECCHRONOS_ID, bindMarker())
+                .value(COLUMN_DC_NAME, bindMarker())
+                .value(COLUMN_NODE_ENDPOINT, bindMarker())
+                .value(COLUMN_NODE_STATUS, bindMarker())
+                .value(COLUMN_LAST_CONNECTION, bindMarker())
+                .value(COLUMN_NEXT_CONNECTION, bindMarker())
+                .value(COLUMN_NODE_ID, bindMarker())
+                .build()
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM));
+        myUpdateStatusStatement = mySession.prepare(QueryBuilder.update(KEYSPACE_NAME, TABLE_NAME)
+                .setColumn(COLUMN_NODE_STATUS, bindMarker())
+                .setColumn(COLUMN_LAST_CONNECTION, bindMarker())
+                .setColumn(COLUMN_NEXT_CONNECTION, bindMarker())
+                .whereColumn(COLUMN_ECCHRONOS_ID).isEqualTo(bindMarker())
+                .whereColumn(COLUMN_DC_NAME).isEqualTo(bindMarker())
+                .whereColumn(COLUMN_NODE_ID).isEqualTo(bindMarker())
+                .build()
+                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM));
+        ecChronosID = builder.myEcchronosID;
+    }
+
+    public void acquireNodes() throws EcChronosException
+    {
+        if (myNodesList.isEmpty())
+        {
+            throw new EcChronosException("Cannot Acquire Nodes because there is no nodes to be acquired");
+        }
+        for (Node node : myNodesList)
+        {
+            LOG.info(
+                    "Preparing to acquire node {} with endpoint {} and Datacenter {}",
+                    node.getHostId(),
+                    node.getEndPoint(),
+                    node.getDatacenter());
+            ResultSet tmpResultSet = acquireNode(node);
+            if (tmpResultSet.wasApplied())
+            {
+                LOG.info("Node successfully acquired by instance {}", ecChronosID);
+            }
+            else
+            {
+                LOG.error("Unable to acquire node {}", node.getHostId());
+            }
+        }
+    }
+
+    private ResultSet acquireNode(final Node node)
+    {
+        return insertNodeInfo(
+                node.getDatacenter(),
+                node.getEndPoint().toString(),
+                node.getState().toString(),
```

Comment:
You are using two different definitions of node status, here when you insert the node you are using the drivers enum NodeState which is "UNKNOWN", "UP", "DOWN" or "FORCED_DOWN". But on line 154 when you update the node status you use the enum NodeStatus which is an ecchronos enum with the values "UNAVAILABLE" or "AVAILABLE".

Reply:
As we discussed, we the generation of the jmx properties should have a try/catch block to avoid any problem changing the node status, also, the loadbalancing policy will be updated by the agent type.